packages/python-editor/src/PythonEditor.tsx

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2018-2021 Elyra Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { ScriptEditor } from '@elyra/script-editor';
+
+import { pyIcon } from '@elyra/ui-components';
+
+import { CodeEditor, IEditorServices } from '@jupyterlab/codeeditor';
+import {
+  ABCWidgetFactory,
+  DocumentRegistry,
+  DocumentWidget
+} from '@jupyterlab/docregistry';
+import { FileEditor } from '@jupyterlab/fileeditor';
+
+export class PythonEditor extends ScriptEditor {
+  /**
+   * Construct a new Python Editor widget.
+   */
+  constructor(
+    options: DocumentWidget.IOptions<FileEditor, DocumentRegistry.ICodeModel>
+  ) {
+    super(options);
+    this.editorLanguage = 'python';
+
+    // Add icon to main tab
+    this.title.icon = pyIcon;
+
+    this.context.ready.then(() => {
+      this.initializeKernelSpecs();
```

Comment:
I suspect both constructors (`PythonEditor` and `REditor`) could be removed simply by treating `ScriptEditor` as an abstract base class that defines two abstract methods: `getLanguage()` and `getIcon()` that `PythonEditor` and `REditor` implement.  Then, wherever there's `this.editorLanguage` in `ScriptEditor`, it is replaced with `this.getLanguage()` and same goes for `this.title.icon` -> `this.getIcon()`.  The intialize Kernelspecs just moves to the `ScriptEditor` constructor as well.

You probably don't need the fields `this.editorLanguage` or `this.title.icon`, although, if they're a hassle to remove, the ScriptEditor constructor could just be like this...

```javascript
    this.editorLanguage = this.getLanguage();

    // Add icon to main tab
    this.title.icon = this.getIcon();

    this.context.ready.then(() => {
      this.initializeKernelSpecs();
```
Since the subclasses know what they are, the fields could probably be made `private`.

I also suspect the factory stuff could be simplified just based on java/python experiences. Typically, the factory for something like this would be a class method on `ScriptEditor` like `ScriptEditor.createInstance()` where `createInstance()` takes an "identifying characteristic" corresponding to the class to create. Language would probably be a good "id".  

Take this with a grain of salt since I don't know how much applies in the JS world.